src/Symfony/Component/Workflow/Workflow.php

```diff
@@ -59,10 +59,6 @@ public function getMarking($subject)
     {
         $marking = $this->markingStore->getMarking($subject);
 
-        if (!$marking instanceof Marking) {
```

Comment:
This is a DX issue. Having an explicit exception is way better than waiting for some random side effect to break the interface in case someone misses the contract. If we need a criteria to decide why the check is desired here but not in other places, I'd say: is it common for users to write their own implementation of the interface ? Then let's do the check. Otherwise it's mostly Symfony internal and no check is fine.